### PR TITLE
Fix isIncrement to return only the version requested

### LINKED ISSUE #27: Entire changelog used when using `--no-increment` flag
Currently when running using the `--no-increment` flag, the entire changelog is provided:
```
# Changelog
All notable changes to this project will be documented in this file.
The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
## [Unreleased]
## [1.0.0-beta.1] - 2022-07-29
- Change1
## [1.0.0-beta.0] - 2022-07-21
- First release
```

This coupled with git.release=true means the github release's content contains the entire changelog.

### Expected
Only the content for the latest version to be included in the release content

### Actual
Entire content of the changelog is contained in the release information

### PR DESCRIPTION
**Pull request by k2snowman69**
Fixes #27

A few issues that made this take longer than I expected:

1. `strictLatest` seems to cause a failure if the changelog only contains Unreleased. There is even [a test case](https://github.com/release-it/keep-a-changelog/blob/master/test.js#L66-L70) which makes me feel this is intended but it really didn't make any sense. I think this is related to https://github.com/release-it/keep-a-changelog/issues/19 and I've marked the lines that need to be removed. I'd like to note, removing these lines however seems to completely eliminate the usage of strictLatest which I'm not sure why it exists
2. The test harness `factory` from `release-it/test/util/index.js` doesn't actually set `isIncrement` if it's set as an option. `isIncrement` is always true no matter what. This is causing the tests to fail. It took me 4 hours to figure out this was my core issue. I spent another hour trying to figure out how to set it and then manually manipulated the variable in memory to make sure my code is good. Without fixing this in the core harness, this code will fail tests.

Some thoughts working in this project

1. The naming of the changelogs isn't clear to what the content is. 
     1. CHANGELOG-FOO - Maybe a better name is "invalid" because it doesn't match the syntax expected?
     2. MISSING - Better name would be CHANGELOG-UNRELEASE_FULL_RELEASE_MISSING
     3. EMPTY - Better name would be CHANGELOG-UNRELEASE_EMPTY_RELEASE_FULL
     4. FULL - Better name would be CHANGELOG-UNRELEASE_FULL_RELEASE_FULL
     5. CHANGELOG-NO-STRICT is the same as CHANGELOG-UNRELEASED
     6. CHANGELOG-FULL only differs from CHANGELOG-UNRELEASED by having a `# Changelog` prefixed
1. Several variations of changelogs aren't being tested such as CHANGELOG-UNRELEASE_EMPTY_RELEASE_EMPTY
1. Tests could use some organizing. Either splitting up the different options into different files or making use of something similar to jest's describe block would help. 
1. A few options are a bit confusing
    1. `keepUnreleased` - I thought this would "keep the unreleased title" in the changelog. However it seems to just not version anything which seems to be the same as isIncrement... so should this be deprecated/removed?
    2. `addUnreleased` - Isn't having an "Unreleased" section part of the https://keepachangelog.com/en/1.0.0/  requirements? So maybe should we just be verifying there is always an unreleased section?

I'm going to stop there, anyway just some thoughts for you.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,37 +53,48 @@ class KeepAChangelog extends Plugin {
...
-
     if (strictLatest && !hasPreviousReleaseSection) {
       throw Error(`Missing section for previous release ("${latestVersion}") in ${filename}.`);
     }
 
-    const startIndex = this.changelogContent.indexOf(this.unreleasedTitle) + this.unreleasedTitle.length;
-    let endIndex = this.changelogContent.indexOf(previousReleaseTitle, startIndex);
-    if (!strictLatest && endIndex === -1) {
-      endIndex = this.changelogContent.length;
+    const { isIncrement } = this.config;
```
Comment (k2snowman69) [inline index.js:66]: This is where the new code starts. Everything between `if (changelog) return changelog;` and `const { isIncrement } = this.config;` is just to satisfy throwing an error if strictLatest is true and the changelog has never had a release before.

File: test.js [hunk 2]
```diff
@@ -56,11 +56,30 @@ test('should throw for empty "unreleased" section', async t => {
   await assert.rejects(runTasks(plugin), /There are no entries under "Unreleased" section in CHANGELOG-EMPTY\.md/);
 });
 
-test('should not throw for empty "unreleased" section when no-increment flag is set', async t => {
-  const options = { increment: false, [namespace]: { filename: 'CHANGELOG-EMPTY.md' } };
+test('should throw for missing "unreleased" section when no-increment flag is set if changelog is misformatted', async t => {
+  const options = { isIncrement: false, [namespace]: { filename: 'CHANGELOG-FOO.md' } };
```
Comment (k2snowman69) [inline test.js]: `isIncrement: false` on these options didn't seem to actually change the value on the config when I debugged the test. I'm not sure how to set this value correctly and i couldn't find any example tests using isDryRun (the only other variable coming off the base config)

Comment (webpro) [reply test.js]: The option to set is `increment: false`, the way to read it from the main config is `this.config.isIncrement`

Comment (k2snowman69) [reply test.js]: Updated! That fixed up all the tests, thanks!

**Comment by webpro** [comment]
Thanks for all the work and the feedback, @k2snowman69! I'm going to look into your feedback seriously, but I don't have the time for this now. Just released this in [v3.1.0](https://github.com/release-it/keep-a-changelog/releases/tag/3.1.0).

**Comment by k2snowman69** [comment]
Glad to help! Let me know if you want to discuss about any of my feedback further. I've got a packed schedule now-a-days but I'll do my best to carve out some time.

**Pull request merged**